browser/ui/views/commands/accelerator_service.cc

```diff
@@ -0,0 +1,188 @@
+#include "base/containers/contains.h"
+#include "base/containers/cxx20_erase_vector.h"
+#include "base/containers/flat_map.h"
+#include "brave/app/command_utils.h"
+#include "brave/components/commands/common/accelerator_parsing.h"
+#include "brave/components/commands/common/accelerator_pref_manager.h"
+#include "brave/components/commands/common/commands.mojom-forward.h"
+#include "brave/components/commands/common/commands.mojom.h"
+#include "components/prefs/pref_service.h"
+#include "mojo/public/cpp/bindings/pending_remote.h"
+#include "ui/base/accelerators/accelerator.h"
+
+namespace commands {
+
+namespace {
+
+mojom::CommandPtr ToMojoCommand(
+    int command_id,
+    const std::vector<ui::Accelerator> accelerators) {
+  auto command = mojom::Command::New();
+  command->id = command_id;
+  command->name = commands::GetCommandName(command_id);
+  for (const auto& accelerator : accelerators) {
+    auto a = mojom::Accelerator::New();
+    a->codes = commands::ToCodesString(accelerator);
+    a->keys = commands::ToKeysString(accelerator);
+    command->accelerators.push_back(std::move(a));
+  }
+  return command;
+}
+
+base::flat_map<int, mojom::CommandPtr> ToMojoCommands(
+    base::flat_map<int, std::vector<ui::Accelerator>> commands) {
+  base::flat_map<int, mojom::CommandPtr> result;
+  for (const auto& [command_id, accelerators] : commands) {
+    result[command_id] = ToMojoCommand(command_id, accelerators);
+  }
+  return result;
+}
+}  // namespace
+
+AcceleratorService::AcceleratorService(PrefService* pref_service,
+                                       const Accelerators& default_accelerators)
+    : pref_manager_(pref_service), default_accelerators_(default_accelerators) {
+  Initialize();
+}
+
+AcceleratorService::~AcceleratorService() = default;
+
+void AcceleratorService::Initialize() {
+  auto accelerators = pref_manager_.GetAccelerators();
+  if (accelerators.size() == 0) {
+    for (const auto& [command_id, default_accelerators] :
+         default_accelerators_) {
+      for (const auto& default_accelerator : default_accelerators) {
+        pref_manager_.AddAccelerator(command_id, default_accelerator);
+      }
+    }
+    accelerators = pref_manager_.GetAccelerators();
+  }
+
+  auto commands = GetCommands();
+  for (const auto& command : commands) {
+    accelerators_.insert({command, {}});
+
+    auto it = accelerators.find(command);
+    if (it == accelerators.end()) {
+      continue;
+    }
+
+    // Initialize the default table.
+    for (const auto& accelerator : it->second) {
+      accelerators_[command].push_back(accelerator);
+    }
+  }
+}
+
+void AcceleratorService::BindInterface(
+    mojo::PendingReceiver<CommandsService> pending_receiver) {
+  receivers_.Add(this, std::move(pending_receiver));
+}
+
+void AcceleratorService::AssignAcceleratorToCommand(
+    int command_id,
+    const std::string& accelerator) {
+  auto a = FromCodesString(accelerator);
+
+  std::vector<int> modified_commands{command_id};
```

Comment:
optional: could you change this so that we could avoid confusion?

```
  std::vector<int> modified_commands = { command_id };
```

It could be just me but initializing `vector<int>` with single argument could look puzzling. 

```
std::vector<int> b{3}; // creates a 1-element vector holding {3}
std::vector<int> a(3); // creates a 3-element vector holding {0, 0, 0}
 
std::vector<int> d{1, 2}; // creates a 2-element vector holding {1, 2}
std::vector<int> c(1, 2); // creates a 1-element vector holding {2}
```

Reply:
Good call!